### DISCUSSION TITLE
How to distinguish between firmware programs of crazyflie-firmware 2.0 and 2.1 when compiling programs ？

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1580] [Q&A]
How to distinguish between firmware programs of crazyflie-firmware 2.0 and 2.1 when compiling programs ？

**Comment by knmcguire** [comment]
Hi!

We need some more info on what you are precisely asking for. Are you referring to the firmware of the latest development branch that you compile locally on your computer? Or how the released firmware reacts depending on the platform?

**Comment by justinleeyang** [comment]
when download nrf51822 and stm32f405 firmware, display as following:

![image](https://github.com/user-attachments/assets/2a0a3d6e-1653-46f0-b8df-dab0dc80f2bc)

but my crazyflie is 2.1, why default detect mpu9250 ? should be BMI088

**Reply by knmcguire** [reply]
You should be seeing the '2.1' indeed here.

Are you using an official stock Crazyflie 2.1?

**Reply by justinleeyang** [reply]
Hi,  
From the crazyflie-firmware code, it is the OTP issue or recompile the firmware, how to write OTP data to stm32 ?

**Reply by knmcguire** [reply]
Could you please answer my question? If this is the case that means that something went wrong during the production testing of the crazyflie and if this is a stock crazyflie, we need to know.

It's fine for you to make your own hardware, but for that we can only give limited support. Please check the discussion rules: https://github.com/orgs/bitcraze/discussions/55

**Reply by justinleeyang** [reply]
Sorry, I'm not sure if it was purchased from the official website. This was left over from a senior at school. I just took it and used it.
